config/7.0/obdemo-bank/ig/scripts/groovy/JwkmsSignClientClaims.groovy

```diff
@@ -27,24 +33,87 @@ if (!attributes.clientCertificate.privateKey) {
     return new Response(Status.BAD_REQUEST)
 }
 
-def jwtClaims = new JwtClaimsSet(request.entity.getJson())
+def optionsHeaders = request.headers.get(routeArgOptionsHeader)
+
+def signingOptions = []
+
+if (optionsHeaders != null) {
+
+    String optionsJson = URLDecoder.decode(optionsHeaders.firstValue.toString())
+    logger.debug("Got options header " + optionsJson)
+    def slurper = new JsonSlurper()
+    signingOptions = slurper.parseText(optionsJson)
+    if (signingOptions == null) {
+        logger.error("Couldn't parse signing options")
+        return new Response(Status.BAD_REQUEST)
+    }
+}
+
 
 PrivateKey signingKey = attributes.clientCertificate.privateKey;
 
 logger.debug("Recovered private key {}",signingKey)
 
-SigningHandler signingHandler = new SigningManager().newRsaSigningHandler(signingKey);
-
+def jwt = ""
+def isDetached = (signingOptions && signingOptions.detachedsig == true)
 def kid = attributes.clientCertificate.serialNumber.toString()
 
-def jwt = new JwtBuilderFactory()
-                .jws(signingHandler)
-                .headers()
-                .alg(JwsAlgorithm.PS256)
-                .kid(kid)
-                .done()
-                .claims(jwtClaims)
-                .build();
+if (signingOptions.unencodedpayload) {
+
+
+    OctetSequenceKey hmacJWK = OctetSequenceKey.parse("{"+
+            "\"kty\":\"oct\"," +
+            "\"k\":\"AyM1SysPpbyDfgZld3umj1qzKObwVMkoqQ-EstJQLr_T-1qS0gZH75aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow\"" +
```

Comment:
Is this temporary?

Reply:
That's left over from initial version - have removed.